pkg/cmd/pr/shared/editable_http.go

```diff
@@ -21,9 +22,29 @@ func UpdateIssue(httpClient *http.Client, repo ghrepo.Interface, id string, isPR
 		return err
 	}
 
-	labelIds, err := options.LabelIds()
-	if err != nil {
-		return err
+	// Labels need to be updated separately to avoid overriding any changes to the list of the
+	// labels from other sources (like the UI).
+	if options.Labels.Edited {
+		if len(options.Labels.Add) > 0 {
+			addedLabelIds, err := options.Metadata.LabelsToIDs(options.Labels.Add)
+			if err != nil {
+				return err
+			}
+			if err := addLabels(httpClient, id, repo, addedLabelIds); err != nil {
+				return err
+			}
+		}
+		if len(options.Labels.Remove) > 0 {
+			removeLabelIds, err := options.Metadata.LabelsToIDs(options.Labels.Remove)
```

Comment:
`addLabels` and `removeLabels` operations are now done sequentially. If they both need to happen, could be do them concurrently? I would typically recommend using `golang.org/x/sync/errgroup` for this, except for the side-effect that a failed HTTP call would automatically cancel all other requests in the group, which is not what we'd want here.

Reply:
I thought about this but ultimately decided not to to avoid doing a partial update in the case that something goes wrong with one of the requests. Seemed better to do an all-or-none update. Idk if there's precedent for doing partial updates in this app. Curious to what your thoughts are about that.

Reply:
> I thought about this but ultimately decided not to to avoid doing a partial update in the case that something goes wrong with one of the requests. Seemed better to do an all-or-none update.

Not sure about what you mean by "partial update", but here is what I was thinking: if I currently do `gh issue edit 123 --title newTitle --add-label bug --remove-label feature`, this will result in 3 separate API requests:
1. `addLabelsToLabelable`
2. `removeLabelsFromLabelable`
3. `updateIssue`/`updatePullRequest`

Right now these 3 API requests will be fired sequentially, but since they are not at all dependent on each other, you could fire them concurrently (e.g. via an `errgroup.Group`) and wait for them all to finish before proceeding. If any of these requests failed, the rest still have a chance of succeeding, and if all of them succeed, the whole `edit` command would take drastically less time overall. In the error scenario, if that is what you meant by "partial update", I think it's fine: the GitHub API doesn't have a functionality to execute a series of updates in a "transaction", so we always risk getting caught in an in-between state. This is expected and shouldn't be an impediment for parallelization.

Reply:
Gotcha. Yeah you understood my point and answered my question. Agree parallelization makes sense here.

Reply:
I implemented parallelism using a regular `WaitGroup`. This way all requests are run even if one fails. I took the liberty of changing the signature of `UpdateIssue` to accept an `io.Writer` for error output. This means all errors can be printed to stderr.